kpet/data.py

```diff
@@ -387,26 +388,105 @@ def validate_host_type_regex(self):
                                   host_type_regex.pattern,
                                   host_types)
 
+    def convert_tree_arches(self):
+        """
+        Get a list of available arches for each tree
+        Returns:
+            Raises a schema.Invalid exception when finding an invalid regex
+        """
+
+        trees = {}
+        # Create a dict {tree: arches,} where arches is a list of regexes
+        for tree in self.trees:
+            trees[tree] = []
+            if "arches" not in self.trees[tree]:
+                self.trees[tree]["arches"] = [".*"]
+
+            for arch_regex in self.trees[tree]["arches"]:
```

Comment:
You can save three lines with this:
```python
    for arch_regex in self.trees.get("arches", [re.compile(".*")]):
```
Maybe put `[re.compile(".*")]` into a variable to avoid compilation on every loop.